salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+    extra_config = None
+
+    def __init__(self, type, path, description, config=None, auth_config=None, extra_config=None):
+        super().__init__(type, path, description, config)
+
+        self.auth_config = auth_config or {}
+        self.extra_config = extra_config or {}
+
+
+class VaultSecretEngine(VaultConfigBase):
+    secret_config = None
+    extra_config = None
+
+    def __init__(self, type, path, description, config=None, secret_config=None, extra_config=None):
+        super().__init__(type, path, description, config)
+
+        self.secret_config = secret_config or {}
+        self.extra_config = extra_config or {}
+
+
+class VaultAuditDevice(VaultConfigBase):
+
+    def __init__(self, type, path, description, config=None):
+        super().__init__(type, path, description, config)
+
+
+class VaultPolicyManager():
+    """Module for handling Vault Policies
+    """
+
+    def __init__(self):
+        """Initialize Vault Policies Manager
+        """
+        log.info("Initializing Vault Policies Manager...")
+
+    def get_remote_policies(self, client, ret):
+        """Retrieve policies from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            [list] -- policies
+        """
+        log.info('Retrieving policies from vault...')
+        polices = []
+        try:
+            policies_resp = client.sys.list_policies()
+
+            for policy in policies_resp['data']['policies']:
+                if not (policy == 'root' or policy == 'default'):
+                    polices.append(policy)
+
+        except Exception:
+            raise
+
+        log.info('Finished retrieving policies from vault.')
+
+        return polices
+
+    def push_policies(self, client, remote_policies, local_policies, ret):
+        """Push policies from local config to remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_policies {list} -- policies from the remote vault server
+            local_policies {list} -- policies from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Pushing policies from local config folder to vault...')
+        new_policies = []
+        try:
+            for policy in local_policies:
+                client.sys.create_or_update_policy(
+                    name=policy['name'],
+                    policy=policy['content']
+                )
+                if policy['name'] in remote_policies:
+                    log.debug('Policy "%s" has been updated.', policy["name"])
+                else:
+                    new_policies.append(policy["name"])
+                    log.debug('Policy "%s" has been created.', policy["name"])
+
+            # Build return object
+            ret['changes']['old'] = remote_policies
+            ret['changes']['new'] = new_policies or "No changes"
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.